### DISCUSSION TITLE
onSuccess, onSettled and onError keep retriggering even with cacheTime set to 0

### DISCUSSION THREAD

**Discussion by danilo-89** [wevm/wagmi#1645] [Q&A]
Is this a bug or I'm doing something wrong?

I am using Vite and useWaitForTransaction hook. 
Whenever hot-reload is initiated and if there is previous data for useWaitForTransaction hook, onSuccess, onSettled and onError will retrigger, even with the cacheTime set to 0. Sometimes it retriggers even without hot-reload.

```
         const {
		data: receiptData,
		isLoading: isReceiptLoading,
		error: receiptError,
		isSuccess: receiptSuccess,
		isFetchedAfterMount: receiptFetchedAfterMount,
		status: receiptStatus,
	} = useWaitForTransaction({
                cacheTime: 0,
		hash: writeData?.hash,
		onSettled(data, error) {
			console.log('Settled', { data, error });
		},
		onSuccess(data) {
			toast.success('Successful', data);
		},
		onError() {
		},
	});
```

**Comment by SaadNiaziDev** [comment]
reset sendTrasaction or contractWrite on page load or when useWaitForTrasaction is settle so  there would be no hash to wait for.
